src/pools/abstract-pool.ts

```diff
@@ -191,9 +190,23 @@ export abstract class AbstractPool<
    * @param worker Workers whose tasks are increased.
    */
   protected increaseWorkersTask (worker: Worker): void {
-    const numberOfTasksTheWorkerHas = this.tasks.get(worker)
-    if (numberOfTasksTheWorkerHas !== undefined) {
-      this.tasks.set(worker, numberOfTasksTheWorkerHas + 1)
+    const numberOfTasksInProgress = this.tasks.get(worker)
+    if (numberOfTasksInProgress !== undefined) {
+      this.tasks.set(worker, numberOfTasksInProgress + 1)
+    } else {
+      throw Error('Worker could not be found in tasks map')
+    }
+  }
+
+  /**
+   * Decrease the number of tasks that the given workers has done.
+   *
+   * @param worker Workers whose tasks are decreased.
+   */
+  protected decreaseWorkersTasks (worker: Worker): void {
+    const numberOfTasksInProgress = this.tasks.get(worker)
+    if (numberOfTasksInProgress !== undefined) {
+      this.tasks.set(worker, numberOfTasksInProgress - 1)
```

Comment:
The code of `increaseWorkersTask` and `decreaseWorkersTasks` is really similar but the `+` or `-`
Could we find a way to merge the together and e.g. pass `+1` or `-1` as a parameter to de-duplicate the code?

Reply:
I will :)

Reply:
I will also add some more test cases since coverage on new code is not so high, as I said I wanted to share this as soon as I had a working version

Reply:
> I will also add some more test cases since coverage on new code is not so high, as I said I wanted to share this as soon as I had a working version

That's the goal of a draft PR with review asked ;)

Reply:
Yes nice feedbacks guys @jerome-benoit  @Shinigami92  I will continue tomorrow :)